src/server/pps/cmds/cmds.go

```diff
@@ -1695,6 +1644,46 @@ func Cmds(mainCtx context.Context, pachCtx *config.Context, pachctlCfg *pachctl.
 	return commands
 }
 
+func setClusterDefaults(ctx context.Context, pachctlCfg *pachctl.Config, r io.ReadCloser, regenerate, reprocess bool) error {
+	b, err := io.ReadAll(r)
+	if err != nil {
+		return errors.Wrap(err, "could not read cluster defaults")
+	}
+	b = bytes.TrimSpace(b) // remove leading & trailing whitespace
+	// validate that the provided defaults parse
+	var cd pps.ClusterDefaults
+	if err := protojson.Unmarshal(b, &cd); err != nil {
+		return errors.Wrapf(err, "invalid cluster defaults")
+	}
+	var req = &pps.SetClusterDefaultsRequest{
+		ClusterDefaultsJson: string(b),
+		Regenerate:          regenerate,
+		Reprocess:           reprocess,
+	}
+
+	client, err := pachctlCfg.NewOnUserMachine(ctx, false)
+	if err != nil {
+		return err
+	}
+	defer client.Close()
+
+	var resp *pps.SetClusterDefaultsResponse
+	if resp, err = client.PpsAPIClient.SetClusterDefaults(ctx, req); err != nil {
```

Comment:
[style question] could you avoid a forward declaration and just do?
```go
resp, err := client.PpsAPIClient.SetClusterDefaults(ctx, req)
if err != nil { ... }
```
not sure which is idiomatic or preferrable here

Reply:
I could. I personally prefer declarations, because it means I can see the exact type, and it means I can use `if resp, err = f(); err != nil { … }` exactly like `if foo, bar.baz, err = f(); err != nil { … }`. I don’t know of a Go style guide which takes a strong stance one way or another.